parser.rb

```diff
@@ -0,0 +1,43 @@
+
+class Parser
+  def initialize(filename)
+    @file = File.open(filename, "r")
+    @text = ''
+    @file.each_line do |txt|
+      @text = @text + txt
+    end
+  end
+
+  def array_of_words
+    @text.split(/\W+/)
```

Comment:
If you wanted to make this work with contractions, you could keep using the split method, but change the regex so that you split on anything that isn't a word character or a single quote. E.g. you could define your regex as follows: `/[^\w']+/`

Alternatively you could look into using the String#scan method, which will create an array of words that match a regex. [link](http://ruby-doc.org/core-2.2.0/String.html)